api/src/test/java/org/apache/iceberg/TestAccessors.java

```diff
@@ -247,4 +252,70 @@ public void testEmptySchema() {
     Schema emptySchema = new Schema();
     assertThat(emptySchema.accessorForField(17)).isNull();
   }
+
+  @Test
+  public void testVariant() throws JsonProcessingException {
+    Base64.Encoder encoder = Base64.getEncoder();
+    boolean expectedTrue = true;
+    boolean expectedFalse = false;
+    int expectedInt = 2147483647;
+    long expectedLong = 2147483648L;
+    float expectedFloat = 1.2345f;
+    double expectedDouble = 1.23456;
+    BigDecimal expectedDecimal = new BigDecimal(123456);
+    String expectedString = "abc";
+    String expectedBytes =
+        new String(encoder.encode(expectedString.getBytes()), StandardCharsets.UTF_8);
+    int nestInt = 10;
+
+    String json =
+        "{\"false\":"
+            + expectedFalse
+            + ", \"true\":"
+            + expectedTrue
+            + ", \"string\": \""
+            + expectedString
+            + "\","
+            + "\"int\":"
+            + expectedInt
+            + ","
+            + "\"long\":"
+            + expectedLong
+            + ", \"float\":"
+            + expectedFloat
+            + ","
+            + "\"double\":"
+            + expectedDouble
+            + ", \"bytes\":\""
+            + expectedBytes
+            + "\", \"decimal\":"
+            + expectedDecimal
+            + ","
+            + "\"nest1\": {\"nest2\":"
+            + nestInt
+            + "}"
+            + "}";
+
+    VariantLike variant = JsonVariant.of(json);
+    assertAccessorReturns(Types.VariantType.get(), variant);
```

Comment:
It looks like this assertion is the only part that's relevant to Accessor tests?

Reply:
I removed JsonVariant for now and also this test now.